client-app/shared/catalog/components/product-sidebar.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="space-y-6">
+    <ProductPriceBlock :product="product">
+      <!-- Price & Delivery (with variations) -->
+      <template v-if="!!product.hasVariations">
+        <div class="flex flex-wrap justify-between gap-x-2 text-base font-bold">
+          <span>
+            {{ $t("pages.product.variations_total_label") }}
+          </span>
+
+          <!-- todo: extract a component for price and use it here -->
+          <span class="text-[--color-success-600]">
+            {{ currentCurrency.symbol }}{{ variationsCartTotalAmount.toFixed(2) }}
+          </span>
+        </div>
+
+        <VcButton class="mt-4 print:hidden" :to="{ name: 'Cart' }" full-width>
+          {{ $t("pages.product.view_cart_button") }}
+        </VcButton>
+      </template>
+
+      <!-- Price & Delivery (without variations) -->
+      <template v-else>
+        <div class="flex flex-wrap justify-between gap-x-2 text-base font-bold">
+          <span>
+            {{ $t("pages.product.price_label") }}
+          </span>
+
+          <VcItemPrice :value="product.price" />
+        </div>
+
+        <div class="mt-4 print:hidden">
+          <AddToCart :product="product" />
+
+          <div class="mt-2 flex">
+            <InStock
+              :is-in-stock="product.availabilityData?.isInStock"
+              :is-digital="isDigital"
+              :quantity="product.availabilityData?.availableQuantity"
+            />
+          </div>
+        </div>
+      </template>
+    </ProductPriceBlock>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { eagerComputed } from "@vueuse/core";
+import { computed } from "vue";
+import { useCurrency } from "@/core/composables";
+import { ProductType } from "@/core/enums";
+import { AddToCart, useCart } from "@/shared/cart";
+import InStock from "./in-stock.vue";
+import ProductPriceBlock from "./product-price-block.vue";
+import type { Product } from "@/core/api/graphql/types";
+
+interface IProps {
+  product: Product;
+}
+
+const props = defineProps<IProps>();
+
+const { currentCurrency } = useCurrency();
+const { getItemsTotal } = useCart();
+
+const isDigital = computed<boolean>(() => props.product.productType === ProductType.Digital);
+
+const variationsCartTotalAmount = eagerComputed<number>(() => {
```

Comment:
eagerComputed for cheep operations, often boolean
here better to use computed
https://v6-7-6.vueuse.org/shared/eagercomputed/